src/main/java/fr/uge/lp/databackbonepoc/controllers/ResearchController.java

```diff
@@ -105,4 +109,23 @@ public ResponseEntity<ResearchSummaryResponse> getResearchSummary(@PathVariable
     var response = ResearchSummaryResponse.from(summary);
     return ResponseEntity.ok(response);
   }
+
+  /**
+   * retrieve the list of ids from left or right rowNumber
+   *
+   * @param researchId
+   * @param valPosition left or right
+   * @return
+   */
+  @GetMapping("/{researchId}/samples/{valPosition}")
+  public ResponseEntity<List<Integer>> getComparisonsLines(
+    @PathVariable @Min(1) Long researchId,
+    @PathVariable @NonNull String valPosition
+  ) {
+    if (! (valPosition.equals("left")||valPosition.equals("right"))){
+      throw new BadRequestException("last parameter needs to be 'left' or 'right'");
```

Comment:
```suggestion
      throw new BadRequestException("valPosition needs to be 'left' or 'right'");
```

C'est mieux de nommer directement le paramètre qui pose problème dans le message d'erreur.